src/components/App.js

```diff
@@ -1,12 +1,43 @@
+import React from 'react';
 import ButtonPanel from './ButtonPanel';
 import Display from './Display';
 import calculate from '../logic/calculate';
 
-const App = () => (
-  <>
-    <Display result="999" />
-    <ButtonPanel />
-  </>
-);
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleClick = this.handleClick.bind(this);
+    this.state = {
+      total: null,
+      next: null,
+      operation: null, // eslint-disable-line react/no-unused-state
+    };
+  }
+
+  handleClick(buttonName) {
+    const { total, next, operation } = this.state;
+    const results = calculate({ total, next, operation }, buttonName);
+    this.setState({
+      total: results.total,
+      next: results.next,
+      operation: results.operation,
+    });
+  }
```

Comment:
__[Optional]__ Use arrow functions to set state. This will enable you to avoid explicit binding in the constructor.